scripts/e2etests/README.md

```diff
@@ -0,0 +1,28 @@
+# NuGet E2E test scripts
+
+This is a collection scripts for Visual Studio End-to-End tests that run via the Package Manager Console.
+
+The End-to-End test depend on an integration provided by the [API.Test](./../../test/TestExtensions/API.Test/API.Test.csproj) project and the Package Manager Console's access to Visual Studio services.
+
+The [CreateEndToEndTestPackage.ps1](./../cibuild/CreateEndToEndTestPackage.ps1) script generates a zip file that contains all the details needed for running End-to-End test. This zip file can also be generated by the [build.ps1](./../../build.ps1) script.
+Note that the scripts assume certain relative paths that are correct within the zip, but may not be work if you attempt to use the scripts from the repo location.
+
+To ensure that all tests are runnable, you may need to update Visual Studio with [tests vs config](build\tests.dev17.vsconfig).
+
+## Running the tests
+
+These tests run on the Visual Studio instance you need to test. They do not need Xunit or any other runner.
+
+1. Create or download the end to end test zip file, and extract it to a directory. It is recommended you extract the files in the `artifacts` directory at the repo root.
+1. Run [SetupFunctionalTests.ps1](./SetupFunctionalTests.ps1). This step is executed once per machine.
+1. Start the VisualStudio instance you want to test. This can be a regular VisualStudio instance, with or without a VSIX that you want to test installed, but it can also be an [experimental Visual Studio instance](./../../docs/debugging.md#debugging-and-testing-nuget-in-visual-studio) you want to debug.
```

Comment:
```suggestion
1. Start the Visual Studio instance you want to test. This can be a regular Visual Studio instance, with or without a VSIX that you want to test installed, but it can also be an [experimental Visual Studio instance](./../../docs/debugging.md#debugging-and-testing-nuget-in-visual-studio) you want to debug.
```